src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -41,36 +50,69 @@ private LogFileGroup(List<LogFile> files, Pattern filePattern, URI directoryURI,
      */
     public static LogFileGroup create(Pattern filePattern, URI directoryURI, Instant lastUpdated)
             throws InvalidLogGroupException {
-        File folder = new File(directoryURI);
+        File logDirectory = new File(directoryURI);
         fileHashToLogFile = new ConcurrentHashMap<>();
 
-        if (!folder.isDirectory()) {
+        if (!logDirectory.isDirectory()) {
             throw new InvalidLogGroupException(String.format("%s must be a directory", directoryURI));
         }
 
-        LogFile[] files = LogFile.of(folder.listFiles());
+        // TODO: Need to make sure hard link dir is empty
+        //  in a less hacky way
+        // TODO: this needs to be in the component work dir, not cwd
         List<LogFile> allFiles = new ArrayList<>();
-        if (files.length != 0) {
-            for (LogFile file: files) {
-                String fileHash = file.hashString();
+        File hardLinkDir = new File(Paths.get(DEFAULT_HARDLINKS_PATH).resolve(logDirectory.getName()).toUri());
+        try {
+            Utils.deleteFileRecursively(hardLinkDir);
```

Comment:
We don't want to call this here then every log group created will delete everything on the hardlinks folder